aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -23,6 +23,16 @@ void bitwise_not_kernel_cuda(TensorIterator& iter) {
   }
 }
 
-REGISTER_DISPATCH(bitwise_not_stub, &bitwise_not_kernel_cuda);
 
+void sign_kernel_cuda(TensorIterator& iter){
+    AT_DISPATCH_ALL_TYPES(iter.dtype(), "sign_cuda", [&]() {
+        gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+            const scalar_t zero = scalar_t(0);
```

Comment:
Could you use `constexpr` instead of `const` so zero gets evaluated during compilation?

Reply:
No... none of this is necessary